service/controller/v12/resource/instance/create.go

```diff
@@ -3,9 +3,7 @@ package instance
 import (
```

Comment:
I refactored this go file to make the EnsureCreated function simpler.
I simply left the basic state-machine logic in that function and moved the code handling each state in a separate function.
I also moved some of the pre-existing functions to different go files in the same module.
Sorry about the ugly diff, but I reckon this refactoring will help writing more unit tests and the general readability of the code